From 4c435469fe5afa569b3239f1fabac1381a5c5d8b Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Fri, 7 Aug 2020 15:12:23 +0000
Subject: [PATCH 04/31] php-ext: replace __GNUC_PREREQ

__GNUC_PREREQ is a thing only present in GNU libc. Replace it by
native GCC-provided macro/identifiers.

Fixes: bp/8.2.0-523-gc70e9c74f
References: KF-3318
---
 php-ext/ECImportContentsChangesProxy.cpp  | 2 +-
 php-ext/ECImportHierarchyChangesProxy.cpp | 2 +-
 php-ext/main.cpp                          | 8 ++------
 php-ext/typeconversion.cpp                | 2 +-
 4 files changed, 5 insertions(+), 9 deletions(-)

diff --git a/php-ext/ECImportContentsChangesProxy.cpp b/php-ext/ECImportContentsChangesProxy.cpp
index 65c7096f6..b134abe2a 100644
--- a/php-ext/ECImportContentsChangesProxy.cpp
+++ b/php-ext/ECImportContentsChangesProxy.cpp
@@ -8,7 +8,7 @@
 #include <cstdio>
 #include <ctime>
 #include <cmath>
-#if __GNUC_PREREQ(5, 0) && !__GNUC_PREREQ(6, 0)
+#if __GNUC__ >= 5 && __GNUC__ < 6 && defined(_GLIBCXX_CMATH)
 using std::isfinite;
 using std::isnan;
 #endif
diff --git a/php-ext/ECImportHierarchyChangesProxy.cpp b/php-ext/ECImportHierarchyChangesProxy.cpp
index 9528e84b7..fccbd73eb 100644
--- a/php-ext/ECImportHierarchyChangesProxy.cpp
+++ b/php-ext/ECImportHierarchyChangesProxy.cpp
@@ -8,7 +8,7 @@
 #include <cstdio>
 #include <ctime>
 #include <cmath>
-#if __GNUC_PREREQ(5, 0) && !__GNUC_PREREQ(6, 0)
+#if __GNUC__ >= 5 && __GNUC__ < 6 && defined(_GLIBCXX_CMATH)
 using std::isfinite;
 using std::isnan;
 #endif
diff --git a/php-ext/main.cpp b/php-ext/main.cpp
index ba20f5c4b..8e8121592 100644
--- a/php-ext/main.cpp
+++ b/php-ext/main.cpp
@@ -50,12 +50,8 @@
 // we need to include this in c++ space because php.h also includes it in
 // 'extern "C"'-space which doesn't work in win32
 #include <cmath>
-#if __GNUC_PREREQ(5, 0) && !__GNUC_PREREQ(6, 0)
-/*
- * Bug/missing feature in preliminary C++ support in GNU libstdc++-v3 5.x;
- * there is also no macro to distinguish between GNU libstdc++ and clang libc++,
- * so that is what you get for clang defining __GNU*.
- */
+#if __GNUC__ >= 5 && __GNUC__ < 6 && defined(_GLIBCXX_CMATH)
+/* Bug/missing feature in preliminary C++ support in GNU libstdc++-v3 5.x */
 using std::isfinite;
 using std::isnan;
 #endif
diff --git a/php-ext/typeconversion.cpp b/php-ext/typeconversion.cpp
index c70cdf025..05cd8bd77 100644
--- a/php-ext/typeconversion.cpp
+++ b/php-ext/typeconversion.cpp
@@ -10,7 +10,7 @@
 #include <mapiutil.h>
 #include <kopano/timeutil.hpp>
 #include "php-ext/phpconfig.h"
-#if __GNUC_PREREQ(5, 0) && !__GNUC_PREREQ(6, 0)
+#if __GNUC__ >= 5 && __GNUC__ < 6 && defined(_GLIBCXX_CMATH)
 using std::isfinite;
 using std::isnan;
 #endif
-- 
2.29.2

